swap_meet/vendor.py

```diff
@@ -1,2 +1,109 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        # If there is no inventory, set self.inventory to an empty list
+        if inventory is None:
+            self.inventory = []
+        # if there is inventory, set it to the provided inventory
+        else: 
+            self.inventory = inventory
+    def add(self, item):
+        # Add the item to the inventory and return the item that was added
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        # try the item exists in the inventory, remove it and return the removed item, else raise error
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_id(self, id):
+        if not isinstance(id, int):
+            raise TypeError("id must be an integer")
+        # Iterate through the inventory to find an item with a matching id
+        for item in self.inventory:  
+            # If an item with a matching id is found, return it  
+            if item.id == id:
+                return item
+        # If no matching item is found, return None
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        # If either the `my_item` is not in this Vendor's inventory or the `their_item` 
+        # is not in the other Vendor's inventory, return False
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        # Remove `my_item` from this Vendor's inventory and add it to the other Vendor's inventory
+        self.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        # Remove `their_item` from the other Vendor's inventory and add it to this Vendor's inventory
+        other_vendor.inventory.remove(their_item)
+        self.add(their_item)
+        # If successful, return True
+        return True
+    
+    # def swap_first_item(self, other_vendor):
+    #     # Check if either the vendor or the friend have empty inventory, and return False if so
+    #     if not self.inventory or not other_vendor.inventory:
+    #         return False
+    #     # Swap the first item from the friend's inventory with the first item from the vendor's inventory
+    #     return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+    def swap_first_item(self, other_vendor):
+        try:
+            # Check if either the vendor or the friend have empty inventory, and return False if so
+            if not self.inventory or not other_vendor.inventory:
+                return False
+            # Swap the first item from the friend's inventory with the first item from the vendor's inventory
+            return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+        except IndexError:
+            return False
```

Comment:
Your guard clause on line 56 should prevent you from hitting any `IndexError`s. I don't think you need the try/except.